src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -538,11 +551,11 @@ void WebRTCTransportProviderServer::HandleProvideOffer(HandlerContext & ctx, con
         }
 
         // Check resource management and stream priorities
-        CHIP_ERROR err = mDelegate.ValidateStreamUsage(req.streamUsage, req.videoStreamID, req.audioStreamID);
+        CHIP_ERROR err = mDelegate.ValidateStreamUsage(req.streamUsage, videoStreamID, audioStreamID);
         if (err != CHIP_NO_ERROR)
         {
-            ChipLogError(Zcl, "HandleProvideOffer: Cannot meet resource management conditions");
-            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+            ChipLogError(Zcl, "HandleProvideOffer: Cannot provide stream usage requested");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::DynamicConstraintError);
```

Comment:
Why return DynamicConstraintError? the spec require return ResourceExhausted here

```
If not able to meet the [Resource Management and Stream Priorities] conditions or are unable to provide another WebRTC session:

Respond with a response status of RESOURCE_EXHAUSTED
```